core/parentchain/block-importer/src/block_importer.rs

```diff
@@ -131,11 +117,11 @@ impl<PB, ValidatorAccessor, OCallApi, StfExecutor, ExtrinsicsFactory, IndirectCa
 	ExtrinsicsFactory: CreateExtrinsics,
 	IndirectCallsExecutor: ExecuteIndirectCalls,
 {
-	type ParentchainBlockType = PB;
+	type SignedBlockType = SignedBlockG<PB>;
 
 	fn import_parentchain_blocks(
 		&self,
-		blocks_to_import: Vec<SignedBlockG<Self::ParentchainBlockType>>,
+		blocks_to_import: Vec<Self::SignedBlockType>,
```

Comment:
Make it as generic as possible

Reply:
The `Block` trait bound makes testing a nightmare. Unless you know of a simple way to construct a signed block in a test environment?

Reply:
What's the problem with creating this?

```
pub struct SignedBlock<Block> {
	/// Full block.
	pub block: Block,
	/// Block justification.
	pub justifications: Option<Justifications>,
}
```

'justifications' is even an option. You can just set it to none.

Reply:
Creating a `Block`, as far as I can tell, is not done so easily - unless there is some shortcut I'm not aware of.

Reply:
Okay I guess with having a `SignedBlock` now, we can more easily mock it. But as soon as we need a real 'Block` it doesn't work as easily anymore?

Reply:
This is the generic block they use in substrate: https://github.com/paritytech/substrate/blob/3fdb445b3b14880017680a3af85e89fb591666a0/primitives/runtime/src/generic/block.rs#L88.

I think substrate has all the necessary stuff for testing with blocks here: https://github.com/paritytech/substrate/blob/master/primitives/runtime/src/testing.rs

Reply:
Ah I just realized the code we're looking at here is not in my new part 😛 My comment is even wrong, it's not made more generic at all. So the whole testing discussion is not really relevant here. But it's good to know about that testing Block, even though I still think it's a hassle to create such a test block.

So in this case here, we still have the `BlockT` trait bound, set on `PB`? I made the change for better readability (not to make it more generic, which it doesn't)

Reply:
Haha, I agree that it is a hassle, but there is just no free lunch. 🥲 

yes, `PB: BlockT` should be the choice.

Reply:
That's the point: `PB` is constraint by `BlockT`: https://github.com/integritee-network/worker/blob/50cf224a057f56629ca3aa8214fd53bdfcc52499/core/parentchain/block-importer/src/parentchain_block_importer.rs#L126 - I haven't changed that.